spec/reporters/abstract_reporter_spec.rb

```diff
@@ -103,11 +103,31 @@
     end
 
     context 'with show_invalid_locations' do
-      let(:reporter) { AbstractReporter.new({show_invalid_locations: true}) }
+      let(:reporter) { AbstractReporter.new(show_invalid_locations: true) }
 
       it 'returns filtered executions' do
-        expect(filtered_executions.size).to eql(20)
-        expect(filtered_executions.first).to eql(invalid_executions.last)
+        expect(filtered_executions.size).to eql(11)
+        expect(filtered_executions.first).to eql(invalid_executions[1])
+      end
+    end
+
+    context 'with unique' do
+      context 'with true' do
+        let(:reporter) { AbstractReporter.new(unique: true) }
+
+        it 'returns filtered executions' do
+          expect(filtered_executions.size).to eql(10)
+          expect(filtered_executions.first).to eql(valid_executions.last)
+        end
+      end
+
+      context 'with false' do
+        let(:reporter) { AbstractReporter.new(unique: false) }
+
+        it 'returns filtered executions' do
+          expect(filtered_executions.size).to eql(10)
```

Comment:
Ideally, we should prepare separated dataset to test `unique`.
However, I'll fix this later.